L1Trigger/L1THGCal/interface/backend/HGCalMulticlusteringImpl.h

```diff
@@ -38,12 +41,38 @@ class HGCalMulticlusteringImpl{
                      const HGCalTriggerGeometryBase & triggerGeometry
                      );
 
+    void clusterizeHistoMax( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr,
+		     l1t::HGCalMulticlusterBxCollection & multiclusters,
+		     const HGCalTriggerGeometryBase & triggerGeometry
+		     );
+
+    void clusterizeHistoThreshold( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr,
+		     l1t::HGCalMulticlusterBxCollection & multiclusters,
+		     const HGCalTriggerGeometryBase & triggerGeometry
+		     );
+
+
 private:
 
     void findNeighbor( const std::vector<std::pair<unsigned int,double>>&  rankedList,
                        unsigned int searchInd,
                        const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtr, 
                        std::vector<unsigned int>& neigbors);
+
+    std::map<std::vector<int>,float> fillHistoClusters( const std::vector<edm::Ptr<l1t::HGCalCluster>> & clustersPtrs );
+
+    std::map<std::vector<int>,float> fillSmoothPhiHistoClusters( std::map<std::vector<int>,float> histoClusters,
```

Comment:
Use const reference for the input parameters instead of passing by value (copying)